### PR TITLE
Samtools sort updata

### PR DESCRIPTION
**Pull request by danilodileo**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/metatdenovo/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/metatdenovo _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Review by erikrikarddaniel — COMMENTED** [review]
Prodigal is failing in the test.

**Comment by danilodileo** [comment]
> Prodigal is failing in the test.

I fixed the issue. During the update it crashed because of the meta.id.

**Comment by danilodileo** [comment]
After the update of some modules I found a bug with transdecoder. The annotation file output (gff3) is not readable by featurecounts . I don't know why it was working before, I am not sure the issue is the file itself, maybe the way that it is named? the gff file has a really long name. I will dig into it more tomorrow as for now I can only work on Rackham and the tests are really slow (because of the queue system)

**Comment by erikrikarddaniel** [comment]
Den tis 11 apr. 2023 kl 13:42 skrev Danilo Di Leo ***@***.***>:
>
> After the update of some modules I found a bug with transdecoder. The annotation file output (gff3) is not readable by featurecounts . I don't know why it was working before, I am not sure the issue is the file itself, maybe the way that it is named? the gff file has a really long name. I will dig into it more tomorrow as for now I can only work on Rackham and the tests are really slow (because of the queue system)
>
OK, I'll look at the PR when you've looked at it.

/D

**Comment by danilodileo** [comment]
> Den tis 11 apr. 2023 kl 13:42 skrev Danilo Di Leo ***@***.***>:
> After the update of some modules I found a bug with transdecoder. The annotation file output (gff3) is not readable by featurecounts . I don't know why it was working before, I am not sure the issue is the file itself, maybe the way that it is named? the gff file has a really long name. I will dig into it more tomorrow as for now I can only work on Rackham and the tests are really slow (because of the queue system)
> OK, I'll look at the PR when you've looked at it. /D
> […](#)

I found how to fix the bug and I am working on it now :D

File: conf/modules.config [hunk 1]
```diff
@@ -257,7 +257,7 @@ process {
     }
 
     withName: '.*:FEATURECOUNTS_CDS' {
-        ext.args = params.orf_caller == 'transdecoder' ? '-g ID -t CDS -F gff3' : '-g ID -t CDS -F gtf'
+        ext.args = '-g ID -t CDS -F gtf'
```
Comment (erikrikarddaniel) [inline conf/modules.config]: Does this work for all three orf callers? It looks like we had a non-functioning lambda before (should have `{}` around it, I think). Don't we need the test anymore, transdecoder outputs the same type of gff as the other two?

**Review by erikrikarddaniel — APPROVED** [review]
If you say tests pass, fine. The config file looks a bit odd though.

File: conf/modules.config (hunk 1 above)
Comment (danilodileo) [reply conf/modules.config]: Yes, I think the issue wasn't the annotation file but a step before. I tested all three orf callers and they work now.

**Pull request merged**